### PR TITLE
Decompile and dedupe dead funcs

### PR DESCRIPTION
**Pull request by Brotenko**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 17 functions (+0.74%) on `us`.

⚠️ This PR introduces 11 warnings:
- src/EA0C10.c:146: warning: initialization from incompatible pointer type
- src/EA0C10.c:252: warning: initialization from incompatible pointer type
- src/EA0C10.c:254: warning: initialization makes pointer from integer without a cast
- src/EA0C10.c:269: warning: implicit declaration of function `func_8004D8E0'
- src/EA0C10.c:296: warning: implicit declaration of function `func_80240DC4_EA16C4'
- src/EA0C10.c:62: warning: initialization from incompatible pointer type
- src/EA8AE0.c:55: warning: initialization makes pointer from integer without a cast
- src/ECAA80.c:139: warning: initialization from incompatible pointer type
- src/ECAA80.c:197: warning: implicit declaration of function `func_80241364_ECB064'
- src/ECAA80.c:219: warning: initialization makes pointer from integer without a cast
- src/world/area_mim/mim_01/B93940.c:36: warning: initialization makes pointer from integer without a cast
- src/world/common/DeadUnkNpcAIMainFunc5.inc.c:8: warning: initialization from incompatible pointer type
- src/world/common/DeadUnkNpcAIMainFunc5.inc.c:8: warning: initialization from incompatible pointer type

File: src/EA0C10.c [hunk 1]
```diff
@@ -56,14 +60,187 @@ s32 N(UnkNpcAIFunc26)(Evt* script) {
...
+        enemy->flags |= ENEMY_FLAGS_100000 | ENEMY_FLAGS_200000 | ENEMY_FLAGS_IGNORE_TOUCH | ENEMY_FLAGS_IGNORE_JUMP | ENEMY_FLAGS_IGNORE_HAMMER | ENEMY_FLAGS_8000000 | ENEMY_FLAGS_10000000;
+        npc->pos.x = 0.0f;
+        npc->pos.y = -1000.0f;
+        npc->pos.z = 0.0f;
+        if (enemy->unk_B0 & 4) {
+            enemy->unk_B0 &= ~4;
+        }
+    }
+
+    func = script->functionTemp[0];
```
Comment (ethteck) [inline src/EA0C10.c:91]: I don't think this should be called func, since it's not a function. If anything, "temp"

File: src/EA0C10.c [hunk 2]
```diff
@@ -56,14 +60,187 @@ s32 N(UnkNpcAIFunc26)(Evt* script) {
...
+                npc->pos.z = posZ;
+                enemy->unk_10.z = npc->pos.z;
+
+                npc->yaw = atan2(npc->pos.x, npc->pos.z, gPlayerStatusPtr->position.x, gPlayerStatusPtr->position.z);
+                enemy->flags &= ~(ENEMY_FLAGS_100000 | ENEMY_FLAGS_IGNORE_TOUCH | ENEMY_FLAGS_IGNORE_JUMP | ENEMY_FLAGS_IGNORE_HAMMER | ENEMY_FLAGS_8000000 | ENEMY_FLAGS_10000000);
+                npc->duration = 0;
+                script->functionTemp[0] = 1;
+            }
+            break;
+
```
Comment (ethteck) [inline src/EA0C10.c:120]: no empty line needed

File: src/EA0C10.c [hunk 3]
```diff
@@ -56,14 +60,187 @@ s32 N(UnkNpcAIFunc26)(Evt* script) {
...
+    }
+
+    return ApiStatus_BLOCK;
+}
 
-INCLUDE_ASM(s32, "EA0C10", func_8024097C_EA127C); // Similar to UnkNpcAIFunc27
+// Similar to UnkNpcAIFunc27
+ApiStatus func_8024097C_EA127C(Evt* script, s32 isInitialCall) {
+    DeadEnemy* enemy = (DeadEnemy*) script->owner1.enemy;
+    Npc *npc = get_npc_unsafe(enemy->npcID);
```
Comment (ethteck) [inline src/EA0C10.c:150]: Pointer on left

File: src/EA0C10.c [hunk 4]
```diff
@@ -56,14 +60,187 @@ s32 N(UnkNpcAIFunc26)(Evt* script) {
...
+            enemy->unk_B0 &= ~4;
+        }
+        enemy->varTable[0] = 0;
+    }
+
+    if (script->functionTemp[0] < 30 && enemy->varTable[0] == 0 && N(UnkNpcAIFunc26)(script)) {
+        script->functionTemp[0] = 30;
+    }
+
+    switch (script->functionTemp[0]) {
```
Comment (ethteck) [inline src/EA0C10.c:193]: no fallthrough comments in this switch block

File: src/world/area_pra/pra_35/D8D270.c [hunk 5]
```diff
@@ -38,8 +38,100 @@ INCLUDE_ASM(s32, "world/area_pra/pra_35/D8D270", func_80241AB4_D8ED04);
...
+    territory.unk_18 = 65.0f;
+    territory.unk_1C = 0;
+
+    if (isInitialCall || (enemy->unk_B0 & 4)) {
+        script->functionTemp[0] = 0;
+        npc->duration = 0;
+        npc->currentAnim.w = enemy->animList[0];
+        npc->flags &= ~0x800;
+        if (!enemy->territory->patrol.isFlying) {
+            npc->flags = (npc->flags | 0x200) & ~0x8;
```
Comment (ethteck) [inline src/world/area_pra/pra_35/D8D270.c:66]: these should be broken into separate statements (|= 0x200, &= ~0x8) (and below too)

File: src/world/area_pra/pra_35/D8D270.c [hunk 6]
```diff
@@ -38,8 +38,100 @@ INCLUDE_ASM(s32, "world/area_pra/pra_35/D8D270", func_80241AB4_D8ED04);
...
+    }
+
+    if ((script->functionTemp[0] < 30) && (enemy->varTable[0] == 0) && N(UnkNpcAIFunc26)(script)) {
+        script->functionTemp[0] = 30;
+    }
+
+    switch (script->functionTemp[0]) {
+        case 0:
+            pra_35_UnkNpcAIFunc24(script, npcAISettings, territoryPtr);
+
```
Comment (ethteck) [inline src/world/area_pra/pra_35/D8D270.c:85]: no empty lines needed, fallthrough comments instead

File: src/world/area_pra/pra_35/D8D270.c [hunk 7]
```diff
@@ -38,8 +38,100 @@ INCLUDE_ASM(s32, "world/area_pra/pra_35/D8D270", func_80241AB4_D8ED04);
...
+            break;
+        case 14:
+            pra_35_UnkNpcDurationFlagFunc(script, npcAISettings, territoryPtr);
+            break;
+        case 30:
+            pra_35_UnkNpcAIFunc6(script);
+        
+        case 31:
+            pra_35_UnkNpcAIFunc7(script);
+            if (script->functionTemp[0] == 32) {
```
Comment (ethteck) [inline src/world/area_pra/pra_35/D8D270.c:118]: the if leaking into other cases is yucky - can you make these like the others?

**Review by ethteck — CHANGES_REQUESTED** [review]

File: include/npc.h [hunk 8]
```diff
@@ -279,6 +279,8 @@ typedef struct Enemy {
     /* 0xDC */ char unk_DC[20];
 } Enemy; // size = 0xF0
 
+
+
```
Comment (JoshDuMan) [inline include/npc.h:283]: No need for these lines?

File: src/world/area_pra/pra_35/D8D270.c (hunk 7 above)
Comment (JoshDuMan) [reply src/world/area_pra/pra_35/D8D270.c:118]: yeah,

if (thing) {
    break;
};

File: src/world/common/DeadUnkNpcAIMainFunc5.inc.c [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+            if (enemy->varTable[7] == 6) {
+                if (rand_int(100) < 33) {
+                    if (enemy->varTable[8] != 0) {
+                        enemy->varTable[8] = 0;
+                        enemy->unk_B5 = 0;
+                        npc->currentAnim.w = 0x55002F;
+                    } else {
+                        enemy->varTable[8] = 1;
+                        enemy->unk_B5 = 1;
+                        npc->currentAnim.w = 0x55002E;
```
Comment (JoshDuMan) [inline src/world/common/DeadUnkNpcAIMainFunc5.inc.c:96]: These anims should have defined values? Don't know what npc offhand

File: src/world/common/DeadUnkNpcAIMainFunc5.inc.c [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+        }
+        if (enemy->varTable[8] != 0) {
+            switch (npc->currentAnim.w - 0x550004) {
+                case 0:
+                case 8:
+                case 10:
+                case 12:
+                case 14:
+                case 18:
+                case 20:
```
Comment (JoshDuMan) [inline src/world/common/DeadUnkNpcAIMainFunc5.inc.c:135]: This switch should remove the - 0x550004 and carry it into each case, using the defined values for anims instead.

File: src/world/common/UnkNpcAIFunc46.inc.c [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+    f32 sp20;
+    f32 sp24;
+    f32 sp28;
+    f32 sp2C;
+    f32 sp30;
+    f32 sp34;
+    
+    npc->duration--;
+    if (enemy->varTable[9] >= npc->duration) {
+        enemy->flags |= 0x1F100000;
```
Comment (JoshDuMan) [inline src/world/common/UnkNpcAIFunc46.inc.c:27]:
```suggestion
        enemy->flags |= (ENEMY_FLAGS_100000 | ENEMY_FLAGS_IGNORE_TOUCH | ENEMY_FLAGS_IGNORE_JUMP | ENEMY_FLAGS_IGNORE_HAMMER | ENEMY_FLAGS_8000000 | ENEMY_FLAGS_10000000);
```

**Review by JoshDuMan — CHANGES_REQUESTED** [review]
Nice job

File: include/dead_structs.h [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    /* 0x0C8 */ s32 unk_C8;
+    /* 0x0CC */ s32* animList;
+    /* 0x0D0 */ EnemyTerritory* territory;
+    /* 0x0D4 */ EnemyDrops* drops;
+    /* 0x0D8 */ u32 tattleMsg;
+    /* 0x0DC */ char unk_DC[42];
+    /* 0x108 */ Vec3f unk_108; // Associated NPC Pos?
+    /* 0x114 */ f32 unk_114;
+    /* 0x118 */ f32 unk_118;
+} DeadEnemy; // size = 0x11C
```
Comment (nanaian) [inline include/dead_structs.h:58]: I think we should be doing this with an `#ifdef DEAD` in the Enemy definition in common_structs.h

File: src/world/common/DeadUnkNpcAIMainFunc5.inc.c (hunk 9 above)
Comment (nanaian) [reply src/world/common/DeadUnkNpcAIMainFunc5.inc.c:96]: see ver/us/build/include/sprite/npc/thrown_bone.h

**Pull request merged**